test/tools/ossfuzz/protomutators/YulProtoMutator.cpp

```diff
@@ -0,0 +1,2081 @@
+static YPR<Block> removeBoundedFor(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				for (auto &stmt: *_message->mutable_statements())
+					if (stmt.has_boundedforstmt())
+					{
+						delete stmt.release_boundedforstmt();
+						stmt.clear_boundedforstmt();
+						break;
+					}
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Remove bounded for statement from statement block"
+		);
+	}
+);
+
+// Add generic for stmt
+static YPR<Block> addGenericFor(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				_message->add_statements()->set_allocated_forstmt(new ForStmt());
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Add for statement to statement block"
+		);
+	}
+);
+
+// Remove generic for stmt
+static YPR<Block> removeGenericFor(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				for (auto &stmt: *_message->mutable_statements())
+					if (stmt.has_forstmt())
+					{
+						delete stmt.release_forstmt();
+						stmt.clear_forstmt();
+						break;
+					}
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Remove for statement from statement block"
+		);
+	}
+);
+
+// Add revert stmt
+static YPR<Block> addRevert(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				auto termStmt = new TerminatingStmt();
+				auto revertStmt = new RetRevStmt();
+				revertStmt->set_stmt(RetRevStmt::REVERT);
+				revertStmt->set_allocated_pos(YPM::litExpression(0));
+				revertStmt->set_allocated_size(YPM::litExpression(0));
+				termStmt->set_allocated_ret_rev(revertStmt);
+				_message->add_statements()->set_allocated_terminatestmt(termStmt);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Add revert(0,0) statement to statement block"
+		);
+	}
+);
+
+// Remove revert statement
+static YPR<Block> removeRevert(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				for (auto &stmt: *_message->mutable_statements())
+					if (stmt.has_terminatestmt() && stmt.terminatestmt().has_ret_rev() &&
+						stmt.terminatestmt().ret_rev().stmt() == RetRevStmt::REVERT)
+					{
+						delete stmt.release_terminatestmt();
+						stmt.clear_terminatestmt();
+						break;
+					}
+			},
+			_message,
+			_seed,
+			YPM::s_lowIP,
+			"Remove revert statement from statement block"
+		);
+	}
+);
+
+// Mutate nullary op
+static YPR<NullaryOp> mutateNullaryOp(
+	[](NullaryOp* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<NullaryOp>(
+			[](NullaryOp* _message, YulRandomNumGenerator& _rand)
+			{
+				_message->clear_op();
+				_message->set_op(
+					YPM::EnumTypeConverter<NullaryOp_NOp>{}.enumFromSeed(_rand())
```

Comment:
How is it related to enums?

Reply:
I didn't understand the question. Just to clarify, this function pseudo randomly chooses an enum from the templated enum list. This is useful for choosing one of several alternative operations e.g., add,subtract, multiply etc.